<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="scider">
<title>A quick start guide to the scider package â€¢ scider</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="A quick start guide to the scider package">
<meta property="og:description" content="scider">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scider</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Quick_start.html">A quick start guide to the scider package</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ChenLaboratory/scider/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>A quick start guide to the scider package</h1>
                        <h4 data-toc-skip class="author">Ning Liu,
Mengbo Li, Yunshun Chen</h4>
            
            <h4 data-toc-skip class="date">2023-09-26</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ChenLaboratory/scider/blob/HEAD/vignettes/Quick_start.Rmd" class="external-link"><code>vignettes/Quick_start.Rmd</code></a></small>
      <div class="d-none name"><code>Quick_start.Rmd</code></div>
    </div>

    
    
<style type="text/css">
  body{
  font-size: 12pt;
}
</style>
<p>scider is an user-friendly R package providing functions to model the
global density of cells in a slide of spatial transcriptomics data. All
functions in the package are built based on the SpatialExperiment
object, allowing integration into various spatial
transcriptomics-related packages from Bioconductor. After modelling
density, the package allows for serveral downstream analysis, including
colocalization analysis, boundary detection analysis and differential
density analysis.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu">install</span><span class="op">(</span><span class="st">"scider"</span><span class="op">)</span></span></code></pre></div>
<p>The development version of <code>scider</code> can be installed from
GitHub:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"ChenLaboratory/scider"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="quick-start">Quick start<a class="anchor" aria-label="anchor" href="#quick-start"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ChenLaboratory/scider" class="external-link">scider</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/drighelli/SpatialExperiment" class="external-link">SpatialExperiment</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-data">Load data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h2>
<p>In this vignette, we will use a subset of a Xenium Breast Cancer
dataset.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"xenium_bc_spe"</span><span class="op">)</span></span></code></pre></div>
<p>In the data, we have quantification of 541 genes from 10000
cells.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SpatialExperiment </span></span>
<span><span class="co">## dim: 541 10000 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(541): ENSG00000121270 ENSG00000213088 ... BLANK_0444</span></span>
<span><span class="co">##   BLANK_0447</span></span>
<span><span class="co">## rowData names(3): ID Symbol Type</span></span>
<span><span class="co">## colnames(10000): cell_212124 cell_120108 ... cell_252054 cell_568560</span></span>
<span><span class="co">## colData names(21): cell_id transcript_counts ... cell_type sample_id</span></span>
<span><span class="co">## reducedDimNames(0):</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span>
<span><span class="co">## spatialCoords names(2) : x_centroid y_centroid</span></span>
<span><span class="co">## imgData names(1): sample_id</span></span></code></pre>
<p>We also have cell-type annotations of these cells, there are 4 cell
types.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">cell_type</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##       B cells Breast cancer   Fibroblasts       T cells </span></span>
<span><span class="co">##           643          3550          4234          1573</span></span></code></pre>
<p>We can use the function <code>plotSpatial</code> to visualise the
cell position and color the cells by cell types.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSpatial.html">plotSpatial</a></span><span class="op">(</span><span class="va">spe</span>, color <span class="op">=</span> <span class="va">cell_type</span>, alpha <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/unnamed-chunk-8-1.png" width="360"></p>
</div>
<div class="section level2">
<h2 id="grid-based-analysis">Grid-based analysis<a class="anchor" aria-label="anchor" href="#grid-based-analysis"></a>
</h2>
<p><code>scider</code> can conduct grid-based density analysis for
spatial transcriptomics data.</p>
<div class="section level3">
<h3 id="density-calculation">Density calculation<a class="anchor" aria-label="anchor" href="#density-calculation"></a>
</h3>
<p>Before calculating density, we need to define cell type-of-interest
(COI). In this case, all cell types are COIs.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">coi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">cell_type</span><span class="op">)</span></span>
<span><span class="va">coi</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Fibroblasts"   "Breast cancer" "T cells"       "B cells"</span></span></code></pre>
<p>We can perform density calculation for each COI using function
<code>gridDensity</code>. The calculated density and grid information
are saved in the metadata of the SpatialExperimnet object.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gridDensity.html">gridDensity</a></span><span class="op">(</span><span class="va">spe</span>, coi <span class="op">=</span> <span class="va">coi</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu">metadata</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "grid_density" "grid_info"</span></span></code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">metadata</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">grid_density</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 11400 rows and 9 columns</span></span>
<span><span class="co">##          x_grid    y_grid    node_x    node_y        node density_fibroblasts</span></span>
<span><span class="co">##       &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt; &lt;integer&gt; &lt;character&gt;           &lt;numeric&gt;</span></span>
<span><span class="co">## 1       329.102   203.852         1         1         1-1          0.00352639</span></span>
<span><span class="co">## 2       329.102   299.810         1         2         1-2          0.00432724</span></span>
<span><span class="co">## 3       329.102   395.767         1         3         1-3          0.00526027</span></span>
<span><span class="co">## 4       329.102   491.725         1         4         1-4          0.00633438</span></span>
<span><span class="co">## 5       329.102   587.683         1         5         1-5          0.00755759</span></span>
<span><span class="co">## ...         ...       ...       ...       ...         ...                 ...</span></span>
<span><span class="co">## 11396   9865.73   10663.2       100       110     100-110          0.00743260</span></span>
<span><span class="co">## 11397   9865.73   10759.2       100       111     100-111          0.00601659</span></span>
<span><span class="co">## 11398   9865.73   10855.2       100       112     100-112          0.00482908</span></span>
<span><span class="co">## 11399   9865.73   10951.1       100       113     100-113          0.00384286</span></span>
<span><span class="co">## 11400   9865.73   11047.1       100       114     100-114          0.00303176</span></span>
<span><span class="co">##       density_breast_cancer density_t_cells density_b_cells</span></span>
<span><span class="co">##                   &lt;numeric&gt;       &lt;numeric&gt;       &lt;numeric&gt;</span></span>
<span><span class="co">## 1               0.000124482      0.00200091      0.00308062</span></span>
<span><span class="co">## 2               0.000158866      0.00269403      0.00447254</span></span>
<span><span class="co">## 3               0.000207521      0.00360691      0.00639036</span></span>
<span><span class="co">## 4               0.000278528      0.00478231      0.00893371</span></span>
<span><span class="co">## 5               0.000385603      0.00624991      0.01215386</span></span>
<span><span class="co">## ...                     ...             ...             ...</span></span>
<span><span class="co">## 11396            0.00992645     1.85862e-05     0.000385502</span></span>
<span><span class="co">## 11397            0.00715312     9.95949e-06     0.000337516</span></span>
<span><span class="co">## 11398            0.00504125     5.53001e-06     0.000288967</span></span>
<span><span class="co">## 11399            0.00348873     3.29144e-06     0.000242472</span></span>
<span><span class="co">## 11400            0.00237981     2.14802e-06     0.000199856</span></span></code></pre>
<p>We can visualise the density of each COI using function
<code>plotDensity</code>.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotDensity.html">plotDensity</a></span><span class="op">(</span><span class="va">spe</span>, coi <span class="op">=</span> <span class="va">coi</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/unnamed-chunk-12-1.png" width="360"></p>
</div>
<div class="section level3">
<h3 id="find-regions-of-interest-rois">Find Regions-of-interest (ROIs)<a class="anchor" aria-label="anchor" href="#find-regions-of-interest-rois"></a>
</h3>
<p>After obtaining grid-based density for each COI, we can then detect
regions-of-interest (ROIs) based on density or select by user.</p>
<div class="section level4">
<h4 id="detected-by-algorithm">Detected by algorithm<a class="anchor" aria-label="anchor" href="#detected-by-algorithm"></a>
</h4>
<p>To detect ROIs automatically, we can use the function
<code>findROI</code>.</p>
<p>The detected ROIs are saved in the metadata of the SpatialExperiment
object.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findROI.html">findROI</a></span><span class="op">(</span><span class="va">spe</span>, coi <span class="op">=</span> <span class="va">coi</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">metadata</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">roi</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 1464 rows and 6 columns</span></span>
<span><span class="co">##      component     members           x           y    xcoord    ycoord</span></span>
<span><span class="co">##       &lt;factor&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">## 1            1      83-107          83         107   8228.13  10375.37</span></span>
<span><span class="co">## 2            1      83-108          83         108   8228.13  10471.33</span></span>
<span><span class="co">## 3            1      84-107          84         107   8324.46  10375.37</span></span>
<span><span class="co">## 4            1      84-108          84         108   8324.46  10471.33</span></span>
<span><span class="co">## 5            1      85-102          85         102   8420.79   9895.58</span></span>
<span><span class="co">## ...        ...         ...         ...         ...       ...       ...</span></span>
<span><span class="co">## 1460        29      53-104          53         104   5338.24   10087.5</span></span>
<span><span class="co">## 1461        29      53-105          53         105   5338.24   10183.5</span></span>
<span><span class="co">## 1462        29      53-106          53         106   5338.24   10279.4</span></span>
<span><span class="co">## 1463        29      54-105          54         105   5434.57   10183.5</span></span>
<span><span class="co">## 1464        29      54-106          54         106   5434.57   10279.4</span></span></code></pre>
<p>We can visualise the ROIs with function <code>plotROI</code>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotROI.html">plotROI</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/unnamed-chunk-14-1.png" width="360"></p>
</div>
<div class="section level4">
<h4 id="select-roi-by-user">Select ROI by user<a class="anchor" aria-label="anchor" href="#select-roi-by-user"></a>
</h4>
<p>Alternatively, users can select ROIs based on their own research
interest (drawn by hand). This can be done using function
<code>selectRegion</code>. This function will open an interactive window
with an interactive plot for users to zoom-in/-out and select ROI using
either a rectangular or lasso selection tool. Users can also press the
<code>Export selected points</code> button to save the ROIs as object in
the R environment.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/selectRegion.html">selectRegion</a></span><span class="op">(</span><span class="fu">metadata</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">grid_density</span>, x.col <span class="op">=</span> <span class="st">"x_grid"</span>, y.col <span class="op">=</span> <span class="st">"y_grid"</span><span class="op">)</span></span></code></pre></div>
<p>After closing the interactive window, the selected ROI has been saved
as a data.frame object named <code>sel_region</code> in the R
environment.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sel_region</span></span></code></pre></div>
<p>We can then use the <code>postSelRegion</code> to save the ROI in the
metadata of the SpatialExperiment object.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/postSelRegion.html">postSelRegion</a></span><span class="op">(</span><span class="va">spe</span>, sel_region <span class="op">=</span> <span class="va">sel_region</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">metadata</span><span class="op">(</span><span class="va">spe1</span><span class="op">)</span><span class="op">$</span><span class="va">roi</span></span></code></pre></div>
<p>Similarly, we can plot visualise the user-defined ROI with function
<code>plotROI</code>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotROI.html">plotROI</a></span><span class="op">(</span><span class="va">spe1</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="testing-relationship-between-cell-types">Testing relationship between cell types<a class="anchor" aria-label="anchor" href="#testing-relationship-between-cell-types"></a>
</h3>
<p>After defining ROIs, we can then test the relationship between any
two cell types within each ROI or overall but account for ROI variation
using a cubic spline or a linear fit.</p>
<p>This can be done with function <code>corrDensity</code>, by setting
the <code>celltype1</code> and <code>celltype2</code> parameters, the
modelling results are saved in the metadata of the SpatialExperiment
object.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">model_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/corDensity.html">corDensity</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span></code></pre></div>
<p>We can see the correlation between breast cancer and fibroblasts in
each ROI.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">model_result</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 174 rows and 9 columns</span></span>
<span><span class="co">##       celltype1     celltype2         ROI     ngrid   cor.coef          t</span></span>
<span><span class="co">##     &lt;character&gt;   &lt;character&gt; &lt;character&gt; &lt;numeric&gt;  &lt;numeric&gt;  &lt;numeric&gt;</span></span>
<span><span class="co">## 1   Fibroblasts Breast cancer           1        25 -0.4212512  -2.227535</span></span>
<span><span class="co">## 2   Fibroblasts Breast cancer           2       106 -0.5961093  -7.571454</span></span>
<span><span class="co">## 3   Fibroblasts Breast cancer           3        27 -0.9058793 -10.694243</span></span>
<span><span class="co">## 4   Fibroblasts Breast cancer           4        42 -0.9245825 -15.348691</span></span>
<span><span class="co">## 5   Fibroblasts Breast cancer           5        36 -0.0392948  -0.229303</span></span>
<span><span class="co">## ...         ...           ...         ...       ...        ...        ...</span></span>
<span><span class="co">## 170     T cells       B cells          25        25   0.459082   2.478267</span></span>
<span><span class="co">## 171     T cells       B cells          26        29   0.865404   8.974262</span></span>
<span><span class="co">## 172     T cells       B cells          27        21  -0.248482  -1.118180</span></span>
<span><span class="co">## 173     T cells       B cells          28        38   0.628826   4.852386</span></span>
<span><span class="co">## 174     T cells       B cells          29        22   0.110237   0.496019</span></span>
<span><span class="co">##            df       p.Pos       p.Neg</span></span>
<span><span class="co">##     &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span><span class="co">## 1          23    0.982010 1.79903e-02</span></span>
<span><span class="co">## 2         104    1.000000 7.83173e-12</span></span>
<span><span class="co">## 3          25    1.000000 4.06755e-11</span></span>
<span><span class="co">## 4          40    1.000000 1.16320e-18</span></span>
<span><span class="co">## 5          34    0.589996 4.10004e-01</span></span>
<span><span class="co">## ...       ...         ...         ...</span></span>
<span><span class="co">## 170        23 1.04871e-02    0.989513</span></span>
<span><span class="co">## 171        27 6.86205e-10    1.000000</span></span>
<span><span class="co">## 172        19 8.61285e-01    0.138715</span></span>
<span><span class="co">## 173        36 1.17842e-05    0.999988</span></span>
<span><span class="co">## 174        20 3.12644e-01    0.687356</span></span></code></pre>
<p>We can also visualise the fitting using function
<code>plotDensCor</code>.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotDensCor.html">plotDensCor</a></span><span class="op">(</span><span class="va">spe</span>, celltype1 <span class="op">=</span> <span class="st">"Breast cancer"</span>, celltype2 <span class="op">=</span> <span class="st">"Fibroblasts"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/unnamed-chunk-21-1.png" width="576"></p>
<p>Or, we can visualise the statistics between cell types using function
<code>plotModStat</code> with a heatmap.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCorHeatmap.html">plotCorHeatmap</a></span><span class="op">(</span><span class="va">model_result</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/unnamed-chunk-22-1.png" width="504"></p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">model_result2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/corDensity.html">corDensity</a></span><span class="op">(</span><span class="va">spe</span>, by.roi <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plotCorHeatmap.html">plotCorHeatmap</a></span><span class="op">(</span><span class="va">model_result2</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/unnamed-chunk-23-1.png" width="360"></p>
</div>
</div>
<div class="section level2">
<h2 id="cell-based-analysis">Cell-based analysis<a class="anchor" aria-label="anchor" href="#cell-based-analysis"></a>
</h2>
<p>Based on the grid density, we can ask many biological question about
the data. For example, we would like to know if a certain cell type that
are located in high density of breast cancer cells are different to the
same cell type from a different level of breast cancer region.</p>
<div class="section level3">
<h3 id="cell-annotation-based-on-grid-density">cell annotation based on grid density<a class="anchor" aria-label="anchor" href="#cell-annotation-based-on-grid-density"></a>
</h3>
<p>To address this question, we first need to divide cells into
different levels of grid density. This can be done using a contour
identification strategy with function <code>getContour</code>.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getContour.html">getContour</a></span><span class="op">(</span><span class="va">spe</span>, coi <span class="op">=</span> <span class="st">"Breast cancer"</span><span class="op">)</span></span></code></pre></div>
<p>Different level of contour can be visualised with cells using
<code>plotContour</code>.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotContour.html">plotContour</a></span><span class="op">(</span><span class="va">spe</span>, coi <span class="op">=</span> <span class="st">"Breast cancer"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/unnamed-chunk-25-1.png" width="360"></p>
<p>We can then annotate cells by their locations within each contour
using function <code>allocateCells</code>.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/allocateCells.html">allocateCells</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSpatial.html">plotSpatial</a></span><span class="op">(</span><span class="va">spe</span>, color <span class="op">=</span> <span class="va">breast_cancer_contour</span>, alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/unnamed-chunk-27-1.png" width="432"></p>
<p>We can visualise cell type composition per level.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCellCompo.html">plotCellCompo</a></span><span class="op">(</span><span class="va">spe</span>, coi <span class="op">=</span> <span class="st">"Breast cancer"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/unnamed-chunk-28-1.png" width="432"></p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCellCompo.html">plotCellCompo</a></span><span class="op">(</span><span class="va">spe</span>, coi <span class="op">=</span> <span class="st">"Breast cancer"</span>, by.roi <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/unnamed-chunk-29-1.png" width="504"></p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] sf_1.0-14                   SpatialExperiment_1.10.0   </span></span>
<span><span class="co">##  [3] SingleCellExperiment_1.22.0 SummarizedExperiment_1.30.2</span></span>
<span><span class="co">##  [5] Biobase_2.60.0              GenomicRanges_1.52.0       </span></span>
<span><span class="co">##  [7] GenomeInfoDb_1.36.3         IRanges_2.34.1             </span></span>
<span><span class="co">##  [9] S4Vectors_0.38.2            BiocGenerics_0.46.0        </span></span>
<span><span class="co">## [11] MatrixGenerics_1.12.3       matrixStats_1.0.0          </span></span>
<span><span class="co">## [13] scider_0.99.0               ggplot2_3.4.3              </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] RColorBrewer_1.1-3        jsonlite_1.8.7           </span></span>
<span><span class="co">##   [3] magrittr_2.0.3            spatstat.utils_3.0-3     </span></span>
<span><span class="co">##   [5] magick_2.7.5              farver_2.1.1             </span></span>
<span><span class="co">##   [7] rmarkdown_2.25            fs_1.6.3                 </span></span>
<span><span class="co">##   [9] zlibbioc_1.46.0           ragg_1.2.5               </span></span>
<span><span class="co">##  [11] vctrs_0.6.3               memoise_2.0.1            </span></span>
<span><span class="co">##  [13] spatstat.explore_3.2-3    DelayedMatrixStats_1.22.6</span></span>
<span><span class="co">##  [15] RCurl_1.98-1.12           janitor_2.2.0            </span></span>
<span><span class="co">##  [17] htmltools_0.5.6           S4Arrays_1.0.6           </span></span>
<span><span class="co">##  [19] Rhdf5lib_1.22.1           rhdf5_2.44.0             </span></span>
<span><span class="co">##  [21] sass_0.4.7                KernSmooth_2.23-21       </span></span>
<span><span class="co">##  [23] bslib_0.5.1               htmlwidgets_1.6.2        </span></span>
<span><span class="co">##  [25] desc_1.4.2                plotly_4.10.2            </span></span>
<span><span class="co">##  [27] lubridate_1.9.2           cachem_1.0.8             </span></span>
<span><span class="co">##  [29] igraph_1.5.1              mime_0.12                </span></span>
<span><span class="co">##  [31] lifecycle_1.0.3           pkgconfig_2.0.3          </span></span>
<span><span class="co">##  [33] Matrix_1.5-4.1            R6_2.5.1                 </span></span>
<span><span class="co">##  [35] fastmap_1.1.1             GenomeInfoDbData_1.2.10  </span></span>
<span><span class="co">##  [37] shiny_1.7.5               snakecase_0.11.1         </span></span>
<span><span class="co">##  [39] digest_0.6.33             colorspace_2.1-0         </span></span>
<span><span class="co">##  [41] rprojroot_2.0.3           tensor_1.5               </span></span>
<span><span class="co">##  [43] dqrng_0.3.1               textshaping_0.3.6        </span></span>
<span><span class="co">##  [45] beachmat_2.16.0           lwgeom_0.2-13            </span></span>
<span><span class="co">##  [47] labeling_0.4.3            fansi_1.0.4              </span></span>
<span><span class="co">##  [49] spatstat.sparse_3.0-2     timechange_0.2.0         </span></span>
<span><span class="co">##  [51] mgcv_1.8-42               httr_1.4.7               </span></span>
<span><span class="co">##  [53] polyclip_1.10-4           abind_1.4-5              </span></span>
<span><span class="co">##  [55] compiler_4.3.1            proxy_0.4-27             </span></span>
<span><span class="co">##  [57] withr_2.5.0               BiocParallel_1.34.2      </span></span>
<span><span class="co">##  [59] DBI_1.1.3                 HDF5Array_1.28.1         </span></span>
<span><span class="co">##  [61] R.utils_2.12.2            DelayedArray_0.26.7      </span></span>
<span><span class="co">##  [63] classInt_0.4-10           rjson_0.2.21             </span></span>
<span><span class="co">##  [65] units_0.8-4               tools_4.3.1              </span></span>
<span><span class="co">##  [67] httpuv_1.6.11             goftest_1.2-3            </span></span>
<span><span class="co">##  [69] R.oo_1.25.0               glue_1.6.2               </span></span>
<span><span class="co">##  [71] nlme_3.1-162              rhdf5filters_1.12.1      </span></span>
<span><span class="co">##  [73] promises_1.2.1            grid_4.3.1               </span></span>
<span><span class="co">##  [75] generics_0.1.3            isoband_0.2.7            </span></span>
<span><span class="co">##  [77] gtable_0.3.4              spatstat.data_3.0-1      </span></span>
<span><span class="co">##  [79] class_7.3-22              R.methodsS3_1.8.2        </span></span>
<span><span class="co">##  [81] tidyr_1.3.0               data.table_1.14.8        </span></span>
<span><span class="co">##  [83] utf8_1.2.3                XVector_0.40.0           </span></span>
<span><span class="co">##  [85] spatstat.geom_3.2-5       pillar_1.9.0             </span></span>
<span><span class="co">##  [87] stringr_1.5.0             limma_3.56.2             </span></span>
<span><span class="co">##  [89] later_1.3.1               splines_4.3.1            </span></span>
<span><span class="co">##  [91] dplyr_1.1.3               lattice_0.21-8           </span></span>
<span><span class="co">##  [93] deldir_1.0-9              tidyselect_1.2.0         </span></span>
<span><span class="co">##  [95] locfit_1.5-9.8            scuttle_1.10.2           </span></span>
<span><span class="co">##  [97] knitr_1.44                edgeR_3.42.4             </span></span>
<span><span class="co">##  [99] xfun_0.40                 DropletUtils_1.20.0      </span></span>
<span><span class="co">## [101] pheatmap_1.0.12           stringi_1.7.12           </span></span>
<span><span class="co">## [103] lazyeval_0.2.2            yaml_2.3.7               </span></span>
<span><span class="co">## [105] evaluate_0.21             codetools_0.2-19         </span></span>
<span><span class="co">## [107] tibble_3.2.1              cli_3.6.1                </span></span>
<span><span class="co">## [109] xtable_1.8-4              systemfonts_1.0.4        </span></span>
<span><span class="co">## [111] munsell_0.5.0             jquerylib_0.1.4          </span></span>
<span><span class="co">## [113] Rcpp_1.0.11               spatstat.random_3.1-6    </span></span>
<span><span class="co">## [115] parallel_4.3.1            ellipsis_0.3.2           </span></span>
<span><span class="co">## [117] pkgdown_2.0.7             sparseMatrixStats_1.12.2 </span></span>
<span><span class="co">## [119] bitops_1.0-7              viridisLite_0.4.2        </span></span>
<span><span class="co">## [121] e1071_1.7-13              scales_1.2.1             </span></span>
<span><span class="co">## [123] purrr_1.0.2               crayon_1.5.2             </span></span>
<span><span class="co">## [125] rlang_1.1.1</span></span></code></pre>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Ning Liu, Mengbo Li, Yunshun Chen.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
