<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="scider">
<title>A quick start guide to the scider package â€¢ scider</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="A quick start guide to the scider package">
<meta property="og:description" content="scider">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scider</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/scider_userGuide.html">A quick start guide to the scider package</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ChenLaboratory/scider/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>A quick start guide to the scider package</h1>
                        <h4 data-toc-skip class="author">Ning Liu,
Mengbo Li, Yunshun Chen</h4>
            
            <h4 data-toc-skip class="date">2024-01-31</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ChenLaboratory/scider/blob/HEAD/vignettes/scider_userGuide.Rmd" class="external-link"><code>vignettes/scider_userGuide.Rmd</code></a></small>
      <div class="d-none name"><code>scider_userGuide.Rmd</code></div>
    </div>

    
    
<style type="text/css">
  body{
  font-size: 12pt;
}
</style>
<p>scider is an user-friendly R package providing functions to model the
global density of cells in a slide of spatial transcriptomics data. All
functions in the package are built based on the SpatialExperiment
object, allowing integration into various spatial
transcriptomics-related packages from Bioconductor. After modelling
density, the package allows for serveral downstream analysis, including
colocalization analysis, boundary detection analysis and differential
density analysis.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu">install</span><span class="op">(</span><span class="st">"scider"</span><span class="op">)</span></span></code></pre></div>
<p>The development version of <code>scider</code> can be installed from
GitHub:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"ChenLaboratory/scider"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="quick-start">Quick start<a class="anchor" aria-label="anchor" href="#quick-start"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ChenLaboratory/scider" class="external-link">scider</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/drighelli/SpatialExperiment" class="external-link">SpatialExperiment</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-data">Load data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h2>
<p>In this vignette, we will use a subset of a Xenium Breast Cancer
dataset.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"xenium_bc_spe"</span><span class="op">)</span></span></code></pre></div>
<p>In the data, we have quantification of 541 genes from 10000
cells.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SpatialExperiment </span></span>
<span><span class="co">## dim: 541 10000 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(541): ENSG00000121270 ENSG00000213088 ... BLANK_0444</span></span>
<span><span class="co">##   BLANK_0447</span></span>
<span><span class="co">## rowData names(3): ID Symbol Type</span></span>
<span><span class="co">## colnames(10000): cell_212124 cell_120108 ... cell_252054 cell_568560</span></span>
<span><span class="co">## colData names(21): cell_id transcript_counts ... cell_type sample_id</span></span>
<span><span class="co">## reducedDimNames(0):</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span>
<span><span class="co">## spatialCoords names(2) : x_centroid y_centroid</span></span>
<span><span class="co">## imgData names(1): sample_id</span></span></code></pre>
<p>We also have cell-type annotations of these cells, there are 4 cell
types.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">cell_type</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##       B cells Breast cancer   Fibroblasts       T cells </span></span>
<span><span class="co">##           643          3550          4234          1573</span></span></code></pre>
<p>We can use the function <code>plotSpatial</code> to visualise the
cell position and color the cells by cell types.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSpatial.html">plotSpatial</a></span><span class="op">(</span><span class="va">spe</span>, color <span class="op">=</span> <span class="va">cell_type</span>, alpha <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span></span></code></pre></div>
<p><img src="scider_userGuide_files/figure-html/unnamed-chunk-8-1.png" width="360"></p>
</div>
<div class="section level2">
<h2 id="grid-based-analysis">Grid-based analysis<a class="anchor" aria-label="anchor" href="#grid-based-analysis"></a>
</h2>
<p><code>scider</code> can conduct grid-based density analysis for
spatial transcriptomics data.</p>
<div class="section level3">
<h3 id="density-calculation">Density calculation<a class="anchor" aria-label="anchor" href="#density-calculation"></a>
</h3>
<p>Before calculating density, we need to define cell type-of-interest
(COI). In this case, all cell types are COIs.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">coi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">cell_type</span><span class="op">)</span></span>
<span><span class="va">coi</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Fibroblasts"   "Breast cancer" "T cells"       "B cells"</span></span></code></pre>
<p>We can perform density calculation for each COI using function
<code>gridDensity</code>. The calculated density and grid information
are saved in the metadata of the SpatialExperimnet object.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gridDensity.html">gridDensity</a></span><span class="op">(</span><span class="va">spe</span>, coi <span class="op">=</span> <span class="va">coi</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu">metadata</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "grid_density" "grid_info"</span></span></code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">metadata</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">grid_density</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 11400 rows and 9 columns</span></span>
<span><span class="co">##          x_grid    y_grid    node_x    node_y        node density_fibroblasts</span></span>
<span><span class="co">##       &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt; &lt;integer&gt; &lt;character&gt;           &lt;numeric&gt;</span></span>
<span><span class="co">## 1       329.102   203.852         1         1         1-1          0.00352639</span></span>
<span><span class="co">## 2       329.102   299.810         1         2         1-2          0.00432724</span></span>
<span><span class="co">## 3       329.102   395.767         1         3         1-3          0.00526027</span></span>
<span><span class="co">## 4       329.102   491.725         1         4         1-4          0.00633438</span></span>
<span><span class="co">## 5       329.102   587.683         1         5         1-5          0.00755759</span></span>
<span><span class="co">## ...         ...       ...       ...       ...         ...                 ...</span></span>
<span><span class="co">## 11396   9865.73   10663.2       100       110     100-110          0.00743260</span></span>
<span><span class="co">## 11397   9865.73   10759.2       100       111     100-111          0.00601659</span></span>
<span><span class="co">## 11398   9865.73   10855.2       100       112     100-112          0.00482908</span></span>
<span><span class="co">## 11399   9865.73   10951.1       100       113     100-113          0.00384286</span></span>
<span><span class="co">## 11400   9865.73   11047.1       100       114     100-114          0.00303176</span></span>
<span><span class="co">##       density_breast_cancer density_t_cells density_b_cells</span></span>
<span><span class="co">##                   &lt;numeric&gt;       &lt;numeric&gt;       &lt;numeric&gt;</span></span>
<span><span class="co">## 1               0.000124482      0.00200091      0.00308062</span></span>
<span><span class="co">## 2               0.000158866      0.00269403      0.00447254</span></span>
<span><span class="co">## 3               0.000207521      0.00360691      0.00639036</span></span>
<span><span class="co">## 4               0.000278528      0.00478231      0.00893371</span></span>
<span><span class="co">## 5               0.000385603      0.00624991      0.01215386</span></span>
<span><span class="co">## ...                     ...             ...             ...</span></span>
<span><span class="co">## 11396            0.00992645     1.85862e-05     0.000385502</span></span>
<span><span class="co">## 11397            0.00715312     9.95949e-06     0.000337516</span></span>
<span><span class="co">## 11398            0.00504125     5.53001e-06     0.000288967</span></span>
<span><span class="co">## 11399            0.00348873     3.29144e-06     0.000242472</span></span>
<span><span class="co">## 11400            0.00237981     2.14802e-06     0.000199856</span></span></code></pre>
<p>We can visualise the density of each COI using function
<code>plotDensity</code>.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotDensity.html">plotDensity</a></span><span class="op">(</span><span class="va">spe</span>, coi <span class="op">=</span> <span class="va">coi</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><img src="scider_userGuide_files/figure-html/unnamed-chunk-12-1.png" width="360"></p>
</div>
<div class="section level3">
<h3 id="find-regions-of-interest-rois">Find Regions-of-interest (ROIs)<a class="anchor" aria-label="anchor" href="#find-regions-of-interest-rois"></a>
</h3>
<p>After obtaining grid-based density for each COI, we can then detect
regions-of-interest (ROIs) based on density or select by user.</p>
<div class="section level4">
<h4 id="detected-by-algorithm">Detected by algorithm<a class="anchor" aria-label="anchor" href="#detected-by-algorithm"></a>
</h4>
<p>To detect ROIs automatically, we can use the function
<code>findROI</code>.</p>
<p>The detected ROIs are saved in the metadata of the SpatialExperiment
object.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findROI.html">findROI</a></span><span class="op">(</span><span class="va">spe</span>, coi <span class="op">=</span> <span class="va">coi</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">metadata</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">roi</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 1627 rows and 6 columns</span></span>
<span><span class="co">##      component     members           x           y    xcoord    ycoord</span></span>
<span><span class="co">##       &lt;factor&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">## 1            1       73-97          73          97   7264.83   9415.79</span></span>
<span><span class="co">## 2            1       73-98          73          98   7264.83   9511.75</span></span>
<span><span class="co">## 3            1       73-99          73          99   7264.83   9607.71</span></span>
<span><span class="co">## 4            1       74-95          74          95   7361.16   9223.88</span></span>
<span><span class="co">## 5            1       74-96          74          96   7361.16   9319.83</span></span>
<span><span class="co">## ...        ...         ...         ...         ...       ...       ...</span></span>
<span><span class="co">## 1623        27      81-109          81         109   8035.47   10567.3</span></span>
<span><span class="co">## 1624        27      82-106          82         106   8131.80   10279.4</span></span>
<span><span class="co">## 1625        27      82-107          82         107   8131.80   10375.4</span></span>
<span><span class="co">## 1626        27      82-108          82         108   8131.80   10471.3</span></span>
<span><span class="co">## 1627        27      82-109          82         109   8131.80   10567.3</span></span></code></pre>
<p>We can visualise the ROIs with function <code>plotROI</code>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotROI.html">plotROI</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span></code></pre></div>
<p><img src="scider_userGuide_files/figure-html/unnamed-chunk-14-1.png" width="360"></p>
</div>
<div class="section level4">
<h4 id="select-roi-by-user">Select ROI by user<a class="anchor" aria-label="anchor" href="#select-roi-by-user"></a>
</h4>
<p>Alternatively, users can select ROIs based on their own research
interest (drawn by hand). This can be done using function
<code>selectRegion</code>. This function will open an interactive window
with an interactive plot for users to zoom-in/-out and select ROI using
either a rectangular or lasso selection tool. Users can also press the
<code>Export selected points</code> button to save the ROIs as object in
the R environment.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/selectRegion.html">selectRegion</a></span><span class="op">(</span><span class="fu">metadata</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">grid_density</span>, x.col <span class="op">=</span> <span class="st">"x_grid"</span>, y.col <span class="op">=</span> <span class="st">"y_grid"</span><span class="op">)</span></span></code></pre></div>
<p>After closing the interactive window, the selected ROI has been saved
as a data.frame object named <code>sel_region</code> in the R
environment.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sel_region</span></span></code></pre></div>
<p>We can then use the <code>postSelRegion</code> to save the ROI in the
metadata of the SpatialExperiment object.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/postSelRegion.html">postSelRegion</a></span><span class="op">(</span><span class="va">spe</span>, sel_region <span class="op">=</span> <span class="va">sel_region</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">metadata</span><span class="op">(</span><span class="va">spe1</span><span class="op">)</span><span class="op">$</span><span class="va">roi</span></span></code></pre></div>
<p>Similarly, we can plot visualise the user-defined ROI with function
<code>plotROI</code>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotROI.html">plotROI</a></span><span class="op">(</span><span class="va">spe1</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="testing-relationship-between-cell-types">Testing relationship between cell types<a class="anchor" aria-label="anchor" href="#testing-relationship-between-cell-types"></a>
</h3>
<p>After defining ROIs, we can then test the relationship between any
two cell types within each ROI or overall but account for ROI variation
using a cubic spline or a linear fit.</p>
<p>This can be done with function <code>corrDensity</code>, by setting
the <code>celltype1</code> and <code>celltype2</code> parameters, the
modelling results are saved in the metadata of the SpatialExperiment
object.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">model_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/corDensity.html">corDensity</a></span><span class="op">(</span><span class="va">spe</span>, trace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>We can see the correlation between breast cancer and fibroblasts in
each ROI.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">model_result</span><span class="op">$</span><span class="va">ROI</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 162 rows and 9 columns</span></span>
<span><span class="co">##       celltype1     celltype2         ROI     ngrid  cor.coef         t</span></span>
<span><span class="co">##     &lt;character&gt;   &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">## 1   Fibroblasts Breast cancer           1        51 -0.261185 -0.726206</span></span>
<span><span class="co">## 2   Fibroblasts Breast cancer           2        64 -0.850429 -3.191725</span></span>
<span><span class="co">## 3   Fibroblasts Breast cancer           3        37 -0.609734 -1.547117</span></span>
<span><span class="co">## 4   Fibroblasts Breast cancer           4       109 -0.619364 -1.301741</span></span>
<span><span class="co">## 5   Fibroblasts Breast cancer           5        81 -0.713589 -2.246380</span></span>
<span><span class="co">## ...         ...           ...         ...       ...       ...       ...</span></span>
<span><span class="co">## 158     T cells       B cells          23        35  0.256837  0.588082</span></span>
<span><span class="co">## 159     T cells       B cells          24        32  0.593693  1.300636</span></span>
<span><span class="co">## 160     T cells       B cells          25        23  0.949763  5.778241</span></span>
<span><span class="co">## 161     T cells       B cells          26        23  0.692722  1.719319</span></span>
<span><span class="co">## 162     T cells       B cells          27        20  0.236879  0.534743</span></span>
<span><span class="co">##            df      p.Pos     p.Neg</span></span>
<span><span class="co">##     &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">## 1     7.20338   0.754685 0.2453153</span></span>
<span><span class="co">## 2     3.89846   0.982807 0.0171933</span></span>
<span><span class="co">## 3     4.04464   0.902016 0.0979838</span></span>
<span><span class="co">## 4     2.72278   0.853903 0.1460965</span></span>
<span><span class="co">## 5     4.86369   0.961945 0.0380552</span></span>
<span><span class="co">## ...       ...        ...       ...</span></span>
<span><span class="co">## 158   4.89692 0.29127441  0.708726</span></span>
<span><span class="co">## 159   3.10775 0.14071356  0.859286</span></span>
<span><span class="co">## 160   3.62552 0.00299803  0.997002</span></span>
<span><span class="co">## 161   3.20414 0.08909362  0.910906</span></span>
<span><span class="co">## 162   4.81012 0.30830407  0.691696</span></span></code></pre>
<p>Or the correlation between breast cancer and fibroblasts across the
whole slide:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">model_result</span><span class="op">$</span><span class="va">overall</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 6 rows and 5 columns</span></span>
<span><span class="co">##       celltype1     celltype2  cor.coef       p.Pos       p.Neg</span></span>
<span><span class="co">##     &lt;character&gt;   &lt;character&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span><span class="co">## 1   Fibroblasts Breast cancer -0.533620 9.99967e-01 6.01577e-08</span></span>
<span><span class="co">## 2   Fibroblasts       T cells  0.593064 1.36192e-07 9.99697e-01</span></span>
<span><span class="co">## 3   Fibroblasts       B cells  0.219668 1.66538e-02 6.99226e-01</span></span>
<span><span class="co">## 4 Breast cancer       T cells -0.633887 9.96068e-01 7.65064e-12</span></span>
<span><span class="co">## 5 Breast cancer       B cells -0.130037 3.96428e-01 1.94609e-03</span></span>
<span><span class="co">## 6       T cells       B cells  0.422464 2.22015e-07 9.88677e-01</span></span></code></pre>
<p>We can also visualise the fitting using function
<code>plotDensCor</code>.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotDensCor.html">plotDensCor</a></span><span class="op">(</span><span class="va">spe</span>, celltype1 <span class="op">=</span> <span class="st">"Breast cancer"</span>, celltype2 <span class="op">=</span> <span class="st">"Fibroblasts"</span><span class="op">)</span></span></code></pre></div>
<p><img src="scider_userGuide_files/figure-html/unnamed-chunk-22-1.png" width="576"></p>
<p>Or, we can visualise the statistics between each pair of cell types
using function <code>plotCorHeatmap</code> in the ROIs:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCorHeatmap.html">plotCorHeatmap</a></span><span class="op">(</span><span class="va">model_result</span><span class="op">$</span><span class="va">ROI</span><span class="op">)</span></span></code></pre></div>
<p><img src="scider_userGuide_files/figure-html/unnamed-chunk-23-1.png" width="504"></p>
<p>Or the correlation between cell type pairs across the whole
slide:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCorHeatmap.html">plotCorHeatmap</a></span><span class="op">(</span><span class="va">model_result</span><span class="op">$</span><span class="va">overall</span><span class="op">)</span></span></code></pre></div>
<p><img src="scider_userGuide_files/figure-html/unnamed-chunk-24-1.png" width="504"></p>
</div>
</div>
<div class="section level2">
<h2 id="cell-based-analysis">Cell-based analysis<a class="anchor" aria-label="anchor" href="#cell-based-analysis"></a>
</h2>
<p>Based on the grid density, we can ask many biological question about
the data. For example, we would like to know if a certain cell type that
are located in high density of breast cancer cells are different to the
same cell type from a different level of breast cancer region.</p>
<div class="section level3">
<h3 id="cell-annotation-based-on-grid-density">cell annotation based on grid density<a class="anchor" aria-label="anchor" href="#cell-annotation-based-on-grid-density"></a>
</h3>
<p>To address this question, we first need to divide cells into
different levels of grid density. This can be done using a contour
identification strategy with function <code>getContour</code>.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getContour.html">getContour</a></span><span class="op">(</span><span class="va">spe</span>, coi <span class="op">=</span> <span class="st">"Breast cancer"</span><span class="op">)</span></span></code></pre></div>
<p>Different level of contour can be visualised with cells using
<code>plotContour</code>.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotContour.html">plotContour</a></span><span class="op">(</span><span class="va">spe</span>, coi <span class="op">=</span> <span class="st">"Breast cancer"</span><span class="op">)</span></span></code></pre></div>
<p><img src="scider_userGuide_files/figure-html/unnamed-chunk-26-1.png" width="360"></p>
<p>We can then annotate cells by their locations within each contour
using function <code>allocateCells</code>.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/allocateCells.html">allocateCells</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSpatial.html">plotSpatial</a></span><span class="op">(</span><span class="va">spe</span>, color <span class="op">=</span> <span class="va">breast_cancer_contour</span>, alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span></code></pre></div>
<p><img src="scider_userGuide_files/figure-html/unnamed-chunk-28-1.png" width="432"></p>
<p>We can visualise cell type composition per level.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCellCompo.html">plotCellCompo</a></span><span class="op">(</span><span class="va">spe</span>, coi <span class="op">=</span> <span class="st">"Breast cancer"</span><span class="op">)</span></span></code></pre></div>
<p><img src="scider_userGuide_files/figure-html/unnamed-chunk-29-1.png" width="432"></p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCellCompo.html">plotCellCompo</a></span><span class="op">(</span><span class="va">spe</span>, coi <span class="op">=</span> <span class="st">"Breast cancer"</span>, by.roi <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="scider_userGuide_files/figure-html/unnamed-chunk-30-1.png" width="504"></p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] sf_1.0-15                   SpatialExperiment_1.12.0   </span></span>
<span><span class="co">##  [3] SingleCellExperiment_1.24.0 SummarizedExperiment_1.32.0</span></span>
<span><span class="co">##  [5] Biobase_2.62.0              GenomicRanges_1.54.1       </span></span>
<span><span class="co">##  [7] GenomeInfoDb_1.38.5         IRanges_2.36.0             </span></span>
<span><span class="co">##  [9] S4Vectors_0.40.2            BiocGenerics_0.48.1        </span></span>
<span><span class="co">## [11] MatrixGenerics_1.14.0       matrixStats_1.2.0          </span></span>
<span><span class="co">## [13] scider_1.1.2                ggplot2_3.4.4              </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] DBI_1.2.1               bitops_1.0-7            deldir_2.0-2           </span></span>
<span><span class="co">##   [4] rlang_1.1.3             magrittr_2.0.3          snakecase_0.11.1       </span></span>
<span><span class="co">##   [7] e1071_1.7-14            compiler_4.3.2          spatstat.geom_3.2-8    </span></span>
<span><span class="co">##  [10] mgcv_1.9-0              systemfonts_1.0.5       vctrs_0.6.5            </span></span>
<span><span class="co">##  [13] stringr_1.5.1           pkgconfig_2.0.3         crayon_1.5.2           </span></span>
<span><span class="co">##  [16] fastmap_1.1.1           magick_2.8.2            XVector_0.42.0         </span></span>
<span><span class="co">##  [19] ellipsis_0.3.2          lwgeom_0.2-13           labeling_0.4.3         </span></span>
<span><span class="co">##  [22] utf8_1.2.4              promises_1.2.1          rmarkdown_2.25         </span></span>
<span><span class="co">##  [25] ragg_1.2.7              purrr_1.0.2             xfun_0.41              </span></span>
<span><span class="co">##  [28] zlibbioc_1.48.0         cachem_1.0.8            jsonlite_1.8.8         </span></span>
<span><span class="co">##  [31] goftest_1.2-3           highr_0.10              later_1.3.2            </span></span>
<span><span class="co">##  [34] DelayedArray_0.28.0     spatstat.utils_3.0-4    R6_2.5.1               </span></span>
<span><span class="co">##  [37] RColorBrewer_1.1-3      bslib_0.6.1             stringi_1.8.3          </span></span>
<span><span class="co">##  [40] spatstat.data_3.0-4     lubridate_1.9.3         jquerylib_0.1.4        </span></span>
<span><span class="co">##  [43] Rcpp_1.0.12             knitr_1.45              tensor_1.5             </span></span>
<span><span class="co">##  [46] splines_4.3.2           igraph_2.0.1            timechange_0.3.0       </span></span>
<span><span class="co">##  [49] httpuv_1.6.14           Matrix_1.6-1.1          tidyselect_1.2.0       </span></span>
<span><span class="co">##  [52] abind_1.4-5             yaml_2.3.8              spatstat.random_3.2-2  </span></span>
<span><span class="co">##  [55] spatstat.explore_3.2-5  lattice_0.21-9          tibble_3.2.1           </span></span>
<span><span class="co">##  [58] shiny_1.8.0             withr_3.0.0             evaluate_0.23          </span></span>
<span><span class="co">##  [61] desc_1.4.3              isoband_0.2.7           units_0.8-5            </span></span>
<span><span class="co">##  [64] proxy_0.4-27            polyclip_1.10-6         pillar_1.9.0           </span></span>
<span><span class="co">##  [67] KernSmooth_2.23-22      plotly_4.10.4           generics_0.1.3         </span></span>
<span><span class="co">##  [70] dbscan_1.1-12           RCurl_1.98-1.14         munsell_0.5.0          </span></span>
<span><span class="co">##  [73] scales_1.3.0            xtable_1.8-4            class_7.3-22           </span></span>
<span><span class="co">##  [76] glue_1.7.0              janitor_2.2.0           pheatmap_1.0.12        </span></span>
<span><span class="co">##  [79] lazyeval_0.2.2          tools_4.3.2             data.table_1.14.10     </span></span>
<span><span class="co">##  [82] fs_1.6.3                grid_4.3.2              tidyr_1.3.1            </span></span>
<span><span class="co">##  [85] colorspace_2.1-0        nlme_3.1-163            GenomeInfoDbData_1.2.11</span></span>
<span><span class="co">##  [88] fastmatrix_0.5-772      cli_3.6.2               spatstat.sparse_3.0-3  </span></span>
<span><span class="co">##  [91] textshaping_0.3.7       fansi_1.0.6             S4Arrays_1.2.0         </span></span>
<span><span class="co">##  [94] viridisLite_0.4.2       dplyr_1.1.4             gtable_0.3.4           </span></span>
<span><span class="co">##  [97] SpatialPack_0.4         sass_0.4.8              digest_0.6.34          </span></span>
<span><span class="co">## [100] classInt_0.4-10         SparseArray_1.2.3       rjson_0.2.21           </span></span>
<span><span class="co">## [103] htmlwidgets_1.6.4       farver_2.1.1            memoise_2.0.1          </span></span>
<span><span class="co">## [106] htmltools_0.5.7         pkgdown_2.0.7           lifecycle_1.0.4        </span></span>
<span><span class="co">## [109] httr_1.4.7              mime_0.12</span></span></code></pre>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Ning Liu, Mengbo Li, Yunshun Chen.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
